### PR TITLE
Engine

### PR DESCRIPTION
**Pull request by szymonjanas**
- added prev/next
- added colorfull consol logs, and file logs,
- refactored engine architecture,
- added sys args,
- added `--help`,
- refactored server architecture,

### CODE REVIEW

**Comment by szymonjanas** [comment]
- added simple README

**Comment by szymonjanas** [comment]
- added travis CI

**Comment by szymonjanas** [comment]
**Done. Review needed!**

File: server/resources/server.html [hunk 1]
```diff
@@ -16,29 +22,35 @@
...
 		</form>
+		<form action="/radio/api/next" method="post">
+			<input class="mediaControlBtn" type="submit" name="nextRadio" value="Next">
+		</form>
+		<form action="/radio/api/prev" method="post">
+			<input class="mediaControlBtn" type="submit" name="prevRadio" value="Prev">
+		</form>
 
-		<form action="/radio/set" method="post">
+		<form action="/radio/api/set" method="post">
```
Comment (piotrlewandowski) [inline server/resources/server.html:38]: I wonder if it makes sense at all to hardcode list of stations in the HTML at all. I'd decouple front end from the backend completely. I'd rather request the radio station via REST API in JSON format? Same goes for other data/actions (like play/stop/next/prev). The way it is built now it refreshes the page every time you do any action and that's not really best way to do modern apps.

**Review by piotrlewandowski — APPROVED** [review]

**Pull request merged**